setuptools/command/bdist_egg.py

```diff
@@ -55,10 +55,11 @@ def write_stub(resource, pyfile):
     _stub_template = textwrap.dedent("""
         def __bootstrap__():
             global __bootstrap__, __loader__, __file__
-            import sys, pkg_resources, imp
+            import sys, pkg_resources
+            from importlib.machinery import ExtensionFileLoader
```

Comment:
`importlib.machinery` is not available on Python 2.7, and although this package has nominally dropped support for Python 2.7, it still maintains compatibility with it. This change will need to be deferred until Python 2.7 support is fully dropped in the codebase... and even then, I'm not sure if this might need to be retained to support building packages that target 2.7.

Reply:
Can we say that it's impossible to invoke `build_ext` on Python 3 to produce artifacts (builds) that are run on Python 2? I think that's the case, but I'm unsure. If we can say, now that the setuptools codebase no longer supports Python 2, that this code will never be run on Python 2, then I'm inclined to accept this change.

Reply:
It seems unlikely to me that this code could appear on Python 2. Let's get it out and see what happens. Worst case, we have to revert it.